cmd/abapEnvironmentPullGitRepo.go

```diff
@@ -0,0 +1,262 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/cookiejar"
+	"os/exec"
+	"time"
+)
+
+func abapEnvironmentPullGitRepo(config abapEnvironmentPullGitRepoOptions) error {
+	r := &runnerExec{}
+	cookieJar, _ := cookiejar.New(nil)
+	client := &http.Client{
+		Jar: cookieJar,
+	}
+
+	var connectionDetails, error = getAbapCommunicationArrangementInfo(config, r)
+	if error != nil {
+		log.Entry().WithError(error).Fatal("Parameters for the ABAP Connection not available")
+		return error
+	}
+
+	var uriConnectionDetails, err = triggerPull(config, connectionDetails, client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("Pull failed on the ABAP System")
+		return err
+	}
+
+	var status, er = pollEntity(config, uriConnectionDetails, client, 10*time.Second)
+	if status == "E" || err != nil {
+		log.Entry().WithError(er).Fatal("Pull failed on the ABAP System")
+		return err
+	}
+
+	return nil
+}
+
+func pollEntity(config abapEnvironmentPullGitRepoOptions, connectionDetails connectionDetailsHTTP, client httpClient, pollIntervall time.Duration) (string, error) {
+
+	log.Entry().Info("Start polling the status...")
+	var status string = "R"
+
+	for {
+		var resp, err = getHTTPResponse("GET", connectionDetails, nil, client)
```

Comment:
as indicated above and for `getHTTPResponse`, please use Piper's `http` package for convenience reasons and for easier testing ...